src/data-access/repositories/cache/cached-rundown-repository.ts

```diff
@@ -26,6 +26,26 @@ export class CachedRundownRepository implements RundownRepository {
     return this.cachedRundowns.get(rundownId) as Rundown
   }
 
+  public getRundownBySegmentId(segmentId: string): Promise<Rundown> {
+    for (const rundown of this.cachedRundowns.values()) {
+      const rundownHasSegment: boolean = rundown.getSegments().some(segment => segment.id === segmentId)
+      if (rundownHasSegment) {
+        return Promise.resolve(rundown)
+      }
+    }
+    throw new NotFoundException(`No Rundown found with a Segment for Segment id: ${segmentId}`)
+  }
+
+  public getRundownByPartId(partId: string): Promise<Rundown> {
```

Comment:
Is it correctly assumed that this is created to handle the database changes and can be removed when sofie server controls ingest?

Reply:
Yes, is so we can find the Rundown a Part belongs to when we get an Ingest update for a Part, since the database only gives us information about the Part.

Whether it can be removed when we control ingest entirely depends on if the Ingest update will give us any information about the Rundown (since we would control it, we would give it some information, so yes, this should be deletable by then).